MetadataExtractor/Formats/QuickTime/QuickTimeMetadataReader.cs

```diff
@@ -113,6 +115,69 @@ void UuidHandler(AtomCallbackArgs a)
                 }
             }
 
+            void MetaDataHandler(AtomCallbackArgs a)
+            {
+                // see https://developer.apple.com/library/archive/documentation/QuickTime/QTFF/Metadata/Metadata.html
+                switch (a.TypeString)
+                {
+                    case "keys":
+                    {
+                        var version = a.Reader.GetByte();
+                        var flags = a.Reader.GetBytes(3);
+                        var entryCount = a.Reader.GetUInt32();
+                        for (int i = 1; i <= entryCount; i++)
+                        {
+                            var keySize = a.Reader.GetUInt32();
+                            var keyValueSize = (int)keySize - 8;
+                            var keyNamespace = a.Reader.GetUInt32();
+                            var keyValue = a.Reader.GetBytes(keyValueSize);
+                            metaDataKeys.Add(Encoding.UTF8.GetString(keyValue));
+                        }
+                        break;
+                    }
+                    case "ilst":
+                    {
+                        var directory = new QuickTimeMetadataHeaderDirectory();
+                        // Iterate over the list of Metadata Item Atoms.
+                        for (int i = 0; i < metaDataKeys.Count; i++)
+                        {
+                            long atomSize = a.Reader.GetUInt32();
+                            if (atomSize < 24)
+                            {
+                                directory.AddError("Invalid ilist AtomSize");
+                                break;
+                            }
+                            var atomType = a.Reader.GetUInt32();
+
+                            // Indexes into the metadata item keys atom are 1-based (1�entry_count).
+                            // atom type for each metadata item atom is the index of the key
+                            if (atomType < 1 || atomType > metaDataKeys.Count)
+                            {
+                                directory.AddError("Invalid ilist AtomType");
+                                break;
+                            }
+                            var key = metaDataKeys[(int)atomType - 1];
+
+                            // Value Atom
+                            var typeIndicator = a.Reader.GetBytes(4);
+                            var localeIndicator = a.Reader.GetBytes(4);
+
+                            // Data Atom
+                            var dataTypeIndicator = a.Reader.GetBytes(4);
+                            var dataLocaleIndicator = a.Reader.GetBytes(4);
+                            var data = a.Reader.GetBytes((int)atomSize - 24);
+                            var dataStr = Encoding.UTF8.GetString(data);
```

Comment:
This should use [`StringValue`](https://github.com/drewnoakes/metadata-extractor-dotnet/blob/master/MetadataExtractor/StringValue.cs) with the encoding defined by the type indicator. This allows a consumer to choose a different encoding at some later point. We regularly see data where the advertised encoding is incorrect, and `StringValue` gives users a way forward.

Reply:
I will do so